src/client/ws/client.rs

```diff
@@ -339,14 +343,33 @@ async fn background_task(
 					break;
 				}
 				Some(Ok(jsonrpc::Response::Single(response))) => {
-					if let Some((unsubscribe, params)) = process_response(&mut manager, response, config.subscription_channel_capacity) {
-						sender.start_request(unsubscribe, params).await.unwrap();
+					match process_response(&mut manager, response, config.subscription_channel_capacity) {
+						Ok(Some((unsubscribe, params))) => {
+							if let Err(e) = sender.start_request(unsubscribe, params).await {
+								log::error!("Failed to send unsubscription response: {:?}", e);
```

Comment:
Note, we could terminate the client here also because the server was probably shut down if the request failed.